web/html/src/components/dialog/LegacyDialog.tsx

```diff
@@ -1,49 +1,72 @@
 import * as React from "react";
-import { PopUp } from "../popup";
-
-// TODO: Move to popup once that gets migrated
-type Instance = JQuery & {
-  modal(...args: any[]): any;
-};
-type Modal = (...args: any[]) => Instance;
+import { useEffect } from "react";
 
 declare global {
   interface JQuery {
-    modal: Modal;
+    modal: (command: "show" | "hide") => JQuery;
   }
 }
 
-export function closeDialog(modalId: string) {
-  const closeModalPromise = new Promise((resolve) => {
-    jQuery("#" + modalId).on("hidden.bs.modal", () => resolve(undefined));
-  });
-  jQuery("#" + modalId).modal("hide");
-  return closeModalPromise;
-}
-
-export type DialogProps = {
+export type LegacyDialogProps = {
   id: string;
+  /** The css className for the 'modal-dialog' div */
   className?: string;
   title?: React.ReactNode;
+  /** The body of the popup */
   content?: React.ReactNode;
-  buttons?: React.ReactNode;
+  /** The footer of the popup */
+  footer?: React.ReactNode;
+  hideHeader?: boolean;
```

Comment:
(Partially out of scope, but now that we are at it....) Do we really use this? And if so, do we really want to use this? I am not sure how good it can be to hide a slice of a proper component design. It does not seem to be a _conditional content_, it affects and hide completely the `header` and the `title` of the box.
If you'd agree but still feel out of scope, feel free to create a separate issue to address this. Thanks

Reply:
I had the same question as you actually, and turns out we use it in `web/html/src/manager/virtualization/guests/console/MessagePopUp.tsx`. I think the idea there is that the doesn't have a title so the header wants to be hidden. I think this prop can be refactored out though if we want, so that the header is hidden automatically if there's no title or header content.

Reply:
Since deprecating this also affected a few other modals due to the way their props were, I decided to leave this be for now.